### ISSUE TITLE
mintlocale

### ISSUE THREAD

**Issue by Aakodal** [NixOS/nixpkgs#121307]
**Project description**
mintlocale is Cinnamon's language helper. It allows users to change their DE's language. Currently, Cinnamon is only available in english under NixOS.

**Metadata**

* homepage URL: https://github.com/linuxmint/mintlocale
* source URL: https://github.com/linuxmint/mintlocale
* license: GPL-2.0
* platforms: linux

**Comment by svavs** [comment]
The Cinnamon interface follows the system locale (i18n.defaultLocale), not exclusively english, if you install the cinnamon.cinnamon-translations package. This worked to me to have the system settings voices in italian.
The mintlocale is helpful to switch between system languages as far as I know.

**Comment by mkg20001** [comment]
@svavs Was installing cinnamon-translations required separately from enabling cinnamon?

I'd have to add it to the module then

@Aakodal Not sure if `mintlocale` should be included, since it goes against nixOS principles.

**Comment by svavs** [comment]
@mkg20001 yes. I've done a first install from the NixOS KDE iso and then, after switching to Cinnamon, I had to add `environment.systemPackages = with pkgs; [ cinnamon.cinnamon-translations ]` to have the package installed and the italian interface. 
Anyway, now I have the control center in my main locale, but the shutdown and disconnect menus are recently again in english, don't know why.

Forgive my ignorance, but for what reason `mintlocale` does not respect the NixOS principles? It's still a useful package to have in Cinnamon (otherwise in the control center is missing the Language section).

**Comment by mkg20001** [comment]
@svavs nixOS is very bold on things being configured through configuration.nix, like `i18n.defaultLocale`. I'm not sure if it is a problem when mintlocale sets the locale for just a specific user, but changing the system locale should only be possible through configuration.nix.

**Comment by svavs** [comment]
@mkg20001 have you any idea why I'm having only the shutdown and disconnect menus in english? All the sistem is translated (Nemo too). Could this be related to the `mintlocale` things?

**Comment by mkg20001** [comment]
The PR for fixing the locale is here https://github.com/NixOS/nixpkgs/pull/124056

Do you mean this UI? Mine seems translated

![image](https://user-images.githubusercontent.com/7735145/119237400-df141800-bb3c-11eb-8b32-49f2561ee1b5.png)

**Comment by svavs** [comment]
> The PR for fixing the locale is here #124056
> 
Thank you! 


> Do you mean this UI? Mine seems translated
>
Yes. The Nemo menu and the desktop icons (see Trash below) are in english too.
This is my situation:
![Schermata del 2021-05-22 22-10-48](https://user-images.githubusercontent.com/1369198/119239841-1807b900-bb4c-11eb-9557-e106ef8cbaf3.png)

**Comment by mkg20001** [comment]
I updated my PR and patched the locale in some places

The display settings and the floppy disk still aren't translated, but I guess that's for another time.

You can apply the patches to your local nixpkgs (in case you use a fork) or use `https://github.com/mkg20001/nixpkgs/archive/cinnamonlocale.tar.gz` as a channel temporarily to get everything translated

![image](https://user-images.githubusercontent.com/7735145/119241544-80a86300-bb57-11eb-974f-0c48ab4c4fac.png)

**Comment by svavs** [comment]
> I updated my PR and patched the locale in some places
> 
Thank you for your work! I'll check it :)

**Comment by mkg20001** [comment]
It appears everything is translated now. In case anyone noticed something not being translated please comment/reopen.

**Comment by svavs** [comment]
Am I missing something in my configuration? I've just upgrated to 21.11 on the stable channel.
![Schermata del 2021-12-06 18-47-00](https://user-images.githubusercontent.com/1369198/144897107-198a25bb-52e0-4b32-b2d5-b42791efe876.png)
.

**Comment by mkg20001** [comment]
Strange. This is only affecting some of the content.
